xla/service/float_normalization.cc

```diff
@@ -290,6 +292,10 @@ Status FloatNormalizationVisitor::ConvertCalledComputations(
   return OkStatus();
 }
 
+bool ShouldSkipCalledComputationsForInstruction(HloInstruction* hlo) {
+  return DynCast<HloAllReduceInstructionBase>(hlo) != nullptr;
```

Comment:
Should we do that for all Collective and not just all-reduce?

Reply:
Maybe. I had two reasons to do it his way:

- IsCollectiveCalledComputation is only set for HloAllReduceInstructionBase instructions and this patch is concerning with fixing the behavior of IsCollectiveCalledComputation.
- I am not sure what is the right predicate for all collectives (I know TJ had some way of doing this, but that actually required fixing the predicate, so I am reluctant to rely on that).

Reply:
I have just switched to explicit opcode, as @cheshire suggested.